### ISSUE TITLE
"func.apply is not a function", when using fullFileBrowser component

### ISSUE THREAD

**Issue by neelduttahere** [TimboKZ/Chonky#121]
Hi, 
As the title mentioned, I am getting the following react library errors, when I use the FullFileBrowser component in my react component. To give a bit more context, my code worked perfectly fine on production level for the past 4 months, but after updating my reactjs version of the project, I started getting this error, where ever I am using the FullFileBrowser component in react.

Below is the series of error:

```
Warning: An effect function must not return anything besides a function, which is used for clean-up. You returned: [object Object]
    in ForwardRef
    in ForwardRef
    in ForwardRef (created by FileBrowser)
    in Unknown (created by FileBrowser)
    in ThemeProvider (created by FileBrowser)
    in ThemeProvider (created by FileBrowser)
    in Provider (created by FileBrowser)
    in IntlProvider (created by FileBrowser)
    in FileBrowser (created by ForwardRef)
    in ForwardRef
    in ForwardRef (at Documents.tsx:755)
    in div (at Documents.tsx:754)
    in div (created by Context.Consumer)
    in div (created by Context.Consumer)
    in Spin (at Documents.tsx:753)
    in Documents (at MissionView.tsx:368)
    in div (created by TabPane)
    in TabPane (at MissionView.tsx:367)
    in div (created by TabPanelList)
    in div (created by TabPanelList)
    in TabPanelList (created by ForwardRef(Tabs))
    in div (created by ForwardRef(Tabs))
    in ForwardRef(Tabs) (created by Context.Consumer)
    in Tabs (at MissionView.tsx:363)
    in DataTabs (at MissionView.tsx:645)
    in div (created by Card)
    in div (created by Card)
    in Card (at MissionView.tsx:1258)
    in div (created by Row)
    in Row (at MissionView.tsx:1257)
    in div (created by Col)
    in Col (at MissionView.tsx:1225)
    in div (created by Row)
    in Row (at MissionView.tsx:1223)
    in div (at MissionView.tsx:1187)
    in MissionView (created by Connect(MissionView))
    in Connect(MissionView) (at Dashboard.tsx:418)
    in Route (at Dashboard.tsx:417)
    in Switch (at Dashboard.tsx:344)
    in ErrorBoundary (at Dashboard.tsx:343)
    in div (at Dashboard.tsx:341)
    in main (created by Basic)
    in Basic (created by Content)
    in Content (at Dashboard.tsx:340)
    in div (created by Col)
    in Col (at Dashboard.tsx:333)
    in div (created by Row)
    in Row (at Dashboard.tsx:126)
    in section (created by BasicLayout)
    in BasicLayout (created by Layout)
    in Layout (at Dashboard.tsx:124)
    in Dashboard (created by Connect(Dashboard))
    in Connect(Dashboard) (at DashboardWrapper.tsx:72)
    in div (at DashboardWrapper.tsx:72)
    in DashboardWrapper (created by Connect(DashboardWrapper))
    in Connect(DashboardWrapper) (at App.tsx:57)
    in Route (at PrivateWrapper.tsx:14)
    in PrivateRoute (created by Connect(PrivateRoute))
    in Connect(PrivateRoute) (at App.tsx:56)
    in Switch (at App.tsx:46)
    in Router (created by BrowserRouter)
    in BrowserRouter (at App.tsx:45)
    in div (at Gmap.tsx:18)
    in Gmap (created by Connect(Gmap))
    in Connect(Gmap) (at App.tsx:43)
    in App (created by Connect(App))
    in Connect(App) (at src/index.tsx:13)
    in Provider (at src/index.tsx:12)
```

```
react-dom.development.js:188 
        
       Uncaught TypeError: func.apply is not a function
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at safelyCallDestroy (react-dom.development.js:19617)
    at react-dom.development.js:20090
    at unstable_runWithPriority (scheduler.development.js:653)
    at runWithPriority$1 (react-dom.development.js:11039)
    at commitUnmount (react-dom.development.js:20083)
    at commitNestedUnmounts (react-dom.development.js:20163)
    at unmountHostComponents (react-dom.development.js:20443)
    at commitDeletion (react-dom.development.js:20500)
    at commitMutationEffects (react-dom.development.js:22782)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at commitRootImpl (react-dom.development.js:22509)
    at unstable_runWithPriority (scheduler.development.js:653)
    at runWithPriority$1 (react-dom.development.js:11039)
    at commitRoot (react-dom.development.js:22381)
    at finishSyncRender (react-dom.development.js:21807)
    at performSyncWorkOnRoot (react-dom.development.js:21793)
    at react-dom.development.js:11089
    at unstable_runWithPriority (scheduler.development.js:653)
    at runWithPriority$1 (react-dom.development.js:11039)
    at flushSyncCallbackQueueImpl (react-dom.development.js:11084)
    at flushSyncCallbackQueue (react-dom.development.js:11072)
    at discreteUpdates$1 (react-dom.development.js:21893)
    at discreteUpdates (react-dom.development.js:806)
    at dispatchDiscreteEvent (react-dom.development.js:4168)
```

```
The above error occurred in the <ForwardRef> component:
    in ForwardRef
    in ForwardRef
    in ForwardRef (created by FileBrowser)
    in Unknown (created by FileBrowser)
    in ThemeProvider (created by FileBrowser)
    in ThemeProvider (created by FileBrowser)
    in Provider (created by FileBrowser)
    in IntlProvider (created by FileBrowser)
    in FileBrowser (created by ForwardRef)
    in ForwardRef
    in ForwardRef (at Documents.tsx:755)
    in div (at Documents.tsx:754)
    in div (created by Context.Consumer)
    in div (created by Context.Consumer)
    in Spin (at Documents.tsx:753)
    in Documents (at MissionView.tsx:368)
    in div (created by TabPane)
    in TabPane (at MissionView.tsx:367)
    in div (created by TabPanelList)
    in div (created by TabPanelList)
    in TabPanelList (created by ForwardRef(Tabs))
    in div (created by ForwardRef(Tabs))
    in ForwardRef(Tabs) (created by Context.Consumer)
    in Tabs (at MissionView.tsx:363)
    in DataTabs (at MissionView.tsx:645)
    in div (created by Card)
    in div (created by Card)
    in Card (at MissionView.tsx:1258)
    in div (created by Row)
    in Row (at MissionView.tsx:1257)
    in div (created by Col)
    in Col (at MissionView.tsx:1225)
    in div (created by Row)
    in Row (at MissionView.tsx:1223)
    in div (at MissionView.tsx:1187)
    in MissionView (created by Connect(MissionView))
    in Connect(MissionView) (at Dashboard.tsx:418)
    in Route (at Dashboard.tsx:417)
    in Switch (at Dashboard.tsx:344)
    in ErrorBoundary (at Dashboard.tsx:343)
    in div (at Dashboard.tsx:341)
    in main (created by Basic)
    in Basic (created by Content)
    in Content (at Dashboard.tsx:340)
    in div (created by Col)
    in Col (at Dashboard.tsx:333)
    in div (created by Row)
    in Row (at Dashboard.tsx:126)
    in section (created by BasicLayout)
    in BasicLayout (created by Layout)
    in Layout (at Dashboard.tsx:124)
    in Dashboard (created by Connect(Dashboard))
    in Connect(Dashboard) (at DashboardWrapper.tsx:72)
    in div (at DashboardWrapper.tsx:72)
    in DashboardWrapper (created by Connect(DashboardWrapper))
    in Connect(DashboardWrapper) (at App.tsx:57)
    in Route (at PrivateWrapper.tsx:14)
    in PrivateRoute (created by Connect(PrivateRoute))
    in Connect(PrivateRoute) (at App.tsx:56)
    in Switch (at App.tsx:46)
    in Router (created by BrowserRouter)
    in BrowserRouter (at App.tsx:45)
    in div (at Gmap.tsx:18)
    in Gmap (created by Connect(Gmap))
    in Connect(Gmap) (at App.tsx:43)
    in App (created by Connect(App))
    in Connect(App) (at src/index.tsx:13)
    in Provider (at src/index.tsx:12)

React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary
```

How to reproduce this?

Call the FullFileBrowser component in you're react code, and compile. On the browser, perform any chonky related action on the chonky file browser component, and you will face this error.

I've tried updating chonky library to the latest version, but the issue still persists. I can't figure out what is causing this, as none of my useEffects are using any callback function, that would trigger an error like this, which based on my research I found that the warning (also pasted in this issue), I am receiveing is due to a purely handled useEffect. I commented out any other useEffects in my script to check whether warning remains, and yes it still remains, but only goes away when I comment out FullFileBrowser component.

**Comment by graywolf336** [comment]
I am running into the same issue here. For me, this is happening with:

`ref = React.createRef<FileBrowserHandle>();`

and then

```
<FullFileBrowser
    ref={this.ref}
```

**---Edit**

Which removing that does not fix it either.

**Comment by neelduttahere** [comment]
Any Idea how to debug this issue from the library? Because this looks a lot like a mishandled useEffect within the FullFileBrowser component.

**Comment by TimboKZ** [comment]
I could not reproduce this locally using the latest version of Create React App, with or without `<React.StrictMode>`. Can one of you provide a minimal setup that reproduces this error?

I did manage to track down one bad `useEffect` instance, which was fixed in https://github.com/TimboKZ/Chonky/commit/5ca86ba130c90c9195e7399ec06a99283982f1ab and released in `chonky@2.3.1`

Please check if you still run into this issue with the most recent version. Feel free to reopen the ticket if the issue persists.

**Comment by neelduttahere** [comment]
Hi,
Surely will check and if the issue still persists after the 5ca86ba commit, I will share a minimal setup to reproduce this issue.

**Comment by neelduttahere** [comment]
@TimboKZ The issue is fixed after your commit 5ca86ba.
